Gtest build failure on Fedora 35 with inbox gcc

### Describe the bug
Gtest build failure on Fedora 35 with inbox gcc
```
 CXX      ucp/gtest-test_ucp_stream.o
../../../test/gtest/uct/uct_test.cc: In member function ‘bool uct_test::async_event_ctx::wait_for_event(uct_test::entity&, double)’:
../../../test/gtest/uct/uct_test.cc:1514:26: error: ‘int poll(pollfd*, nfds_t, int)’ writing 8 bytes into a region of size 4 overflows the destination [-Werror=stringop-overflow=]
 1514 |     int ret        = poll(&wakeup_fd, 1, timeout_ms);
      |                      ~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/poll.h:1,
                 from ../../../test/gtest/uct/uct_test.h:15,
                 from ../../../test/gtest/uct/uct_test.cc:7:
/usr/include/sys/poll.h:38:9: note: destination object ‘pollfd::fd’ of size 4
   38 |     int fd;                     /* File descriptor to poll.  */
      |         ^~
/usr/include/sys/poll.h:38:9: note: destination object ‘pollfd::fd’ of size 4
/usr/include/sys/poll.h:54:12: note: in a call to function ‘int poll(pollfd*, nfds_t, int)’ declared with attribute ‘access (write_only, 1, 2)’
   54 | extern int poll (struct pollfd *__fds, nfds_t __nfds, int __timeout)
      |            ^~~~
  CXX      ucp/gtest-test_ucp_peer_failure.o

```

### Steps to Reproduce
- ``../configure --enable-gtest --enable-examples --with-valgrind --enable-profiling --enable-frame-pointer --enable-stats --enable-memtrack --enable-fault-injection --enable-debug-data --enable-mt --with-xpmem=/home/pavsha01/work/xpmem-install/ --prefix=/home/xxx/xxx/ucx-install/ --with-ugni PKG_CONFIG_PATH=:/home/xxx/xxx/ucx/build/../contrib/cray-ugni-mock PKG_CONFIG_TOP_BUILD_DIR=/home/xxx/xxx/ucx/build/..``

- Reproducible on master and v1.12.x branch


### Setup and versions
- Fedora release 35 (Thirty Five) , aarch64
- No rdma-core drivers/hardware
- gcc (GCC) 11.2.1 20210728 (Red Hat 11.2.1-1)

--- a/test/gtest/uct/uct_test.cc
+++ b/test/gtest/uct/uct_test.cc
@@ -1494,32 +1494,37 @@ void uct_test::async_event_ctx::signal() {
 }
 
 bool uct_test::async_event_ctx::wait_for_event(entity &e, double timeout_sec) {
-    if (wakeup_fd.fd == -1) {
+    /* Caching fd locally to workaround bug in gcc (GCC) 11.2.1 20210728 FC35
+     * See details in https://github.com/openucx/ucx/issues/7705 */
+    struct pollfd local_fd = wakeup_fd;
+
+    if (local_fd.fd == -1) {
         /* create wakeup */
         if (e.iface_attr().cap.event_flags & UCT_IFACE_FLAG_EVENT_FD) {
-            ucs_status_t status =
-                uct_iface_event_fd_get(e.iface(), &wakeup_fd.fd);
+            ucs_status_t status = uct_iface_event_fd_get(e.iface(),
+                                                         &local_fd.fd);
             ASSERT_UCS_OK(status);
         } else {
-            ucs_status_t status =
-                ucs_async_pipe_create(&aux_pipe);
+            ucs_status_t status = ucs_async_pipe_create(&aux_pipe);
             ASSERT_UCS_OK(status);
             aux_pipe_init = true;
-            wakeup_fd.fd = ucs_async_pipe_rfd(&aux_pipe);
+            local_fd.fd   = ucs_async_pipe_rfd(&aux_pipe);
         }
     }
 
     int timeout_ms = static_cast<int>((timeout_sec * UCS_MSEC_PER_SEC) *
                                       ucs::test_time_multiplier());
-    int ret        = poll(&wakeup_fd, 1, timeout_ms);
+    int ret        = poll(&local_fd, 1, timeout_ms);
     EXPECT_TRUE((ret == 0) || (ret == 1));
     if (ret > 0) {
         if (e.iface_attr().cap.event_flags & UCT_IFACE_FLAG_EVENT_ASYNC_CB) {
             ucs_async_pipe_drain(&aux_pipe);
         }
+        wakeup_fd = local_fd;
         return true;
     }
 
+    wakeup_fd = local_fd;
     return false;
 }
 

